src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -696,9 +696,17 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                     handleOne(callbacks, idx + 1);
                 } else if (callback instanceof AuthorizeCallback) {
                     final AuthorizeCallback authorizeCallback = (AuthorizeCallback) callback;
+                    String authenticationID = authorizeCallback.getAuthenticationID();
+                    if (authenticationID == null) {
+                        authenticationID = getAuthenticationPrincipal().getName();
+                    }
                     // always re-set the authentication name to ensure it hasn't changed.
-                    setAuthenticationName(authorizeCallback.getAuthenticationID());
-                    authorizeCallback.setAuthorized(authorize(authorizeCallback.getAuthorizationID()));
+                    setAuthenticationName(authenticationID);
+                    String authorizationID = authorizeCallback.getAuthorizationID();
+                    if (authorizationID == null) {
+                        authorizationID = authenticationID;
```

Comment:
This also seems wrong.  If the authorizationID is null, then we should call ``ServerAuthenticationContext#authorize()``; if it is not null, then we should call ``ServerAuthenticationContext#authorize(java.lang.String)`` with the name being authorized.